secrets/awskms/kms.go

```diff
@@ -98,9 +99,28 @@ func (k *keeper) ErrorAs(err error, i interface{}) bool {
 }
 
 // ErrorCode implements driver.ErrorCode.
-func (k *keeper) ErrorCode(error) gcerrors.ErrorCode {
-	// TODO(shantuo): try to classify aws error codes
-	return gcerrors.Unknown
+func (k *keeper) ErrorCode(err error) gcerrors.ErrorCode {
+	ae, ok := err.(awserr.Error)
+	if !ok {
+		return gcerr.Unknown
+	}
+	ec, ok := errorCodeMap[ae.Code()]
+	if !ok {
+		return gcerr.Unknown
+	}
+	return ec
+}
+
+var errorCodeMap = map[string]gcerrors.ErrorCode{
```

Comment:
Nit: can you organize these by value? So all of the `gcerrors.PermissionDenied` keys are grouped together. I think that makes it easier to read.

Reply:
Done.